flash/data/process.py

```diff
@@ -101,6 +101,164 @@ class PreprocessState:
 
 
 class Preprocess(Properties, torch.nn.Module):
+    """
+    The :class:`~flash.data.process.Preprocess` is used to encapsulate ``all the processing logic`` from data loading up to the model.
+
+    It is particularly relevant when you want to provide ready-to-production implementation which works
+    with 4 different stages: ``TRAINING``, ``VALIDATING``, ``TESTING``, ``PREDICTING``.
+
+    The :class:`~flash.data.process.Preprocess` is currently supporting the following hooks:
+
+        - ``load_data``: Expect some metadata and return an Mapping (can be a ``Dataset``, but not recommended)
```

Comment:
Not clear what is "some metadata" and what mapping. can we be specific?

Reply:
Added an example ? Is it better ?